integration_test/auth.integration.test.js

```diff
@@ -0,0 +1,238 @@
+const { test } = require('ava')
+const auth = require('./util/auth')
+const axios = require('axios')
+const gqls = require('./auth.integration.test.gql')
+
+const context = {
+  helper: undefined,
+  testNote: 'auth, inmem'
+}
+
+async function authenticate (test, username, password) {
+  test.log(`Authenticating as ${username}`)
+  const authHeaders = await auth.authenticateKeycloak(username, password)
+  context.helper.resetApolloClient(authHeaders)
+  test.log(`Authenticated as ${username}`)
+}
+
+test.before(async t => {
+  process.env.KEYCLOAK_CONFIG_FILE = require('path').resolve('./keycloak/keycloak.json')
```

Comment:
We should not use the same realm for tests. Let's create separate realm as any changes and improvements can possibly break this test un future. With changes we have there if server goes down we are losing all the work. 

- We should have separate realm
- We should have separate client config

Reply:
ok, good idea, I also don't know how this impacts security. I think we can use credentials passed by the environment into CircleCI and separate realm.

Reply:
@wojta Not sure what you mean, but to answer this generally: 
we can use predefined users and map them into code in integration tests - there is no need for CircleCi to be aware of anything.